### DISCUSSION TITLE
How to 'destroy' entities

### DISCUSSION THREAD

**Discussion by ghost** [stride3d/stride#1349] [Q&A] [answered]
i'm trying to remove an object by calling entity.Destroy(); but that says "'ComponentBase.Destroy()' is inaccessible due to its protection level" so i'm not sure how to call that method? i come from python, so i'm having trouble understanding what the keywords and classes mean.

**Comment by Eideren** [comment, marked as answer]
Entities aren't manually destroyed in stride, they act like any reference type so they will be 'destroyed' automatically by the garbage collector once nothing references them anymore.
Destroy in this case is a protected method used by the engine in certain circumstances.
How about we continue this in [your previous thread](https://github.com/stride3d/stride/discussions/1348) though, it's the same issue.
